pecos/xmc/xlinear/model.py

```diff
@@ -299,6 +299,53 @@ def split_model_at_depth(self, given_depth, reindex=False):
             )
         return split_model
 
+    @classmethod
+    def reconstruct_model(cls, parent_model, child_models, Y_ids_of_subtrees):
+        """Reconstruct the model with parent_model and child_models
+
+        Args:
+            parent_model: a XLinearModel that has model_chain of the
+                    top part of original model
+            child_models: A list of child models
+            Y_ids_of_subtrees: A list of index mapping to original labels from the child model output
+
+        Returns:
+            model: a XLinearModel that conposed by parent_model and child_models
+
+        Notes:
+            As am example if parent_model is 2*1 XLinearModel, child_models are two: 2*1 XLinearModel,
+            and Y_ids_of_subtrees is [[0,1], [2,3]]
+            After reconstruction, we will have a model that is a complete binary tree of depth 2, where
+            its first layer is same as parent models and the second layer is same as the concat of two child_models.
+        """
+        permuted_Y_label = np.concatenate(Y_ids_of_subtrees)
```

Comment:
Add check that `child_models` having same length as `Y_ids_of_subtrees`. Also, should we add default for `Y_ids_of_subtrees=None` and concat the models in given order?